src/sentry/api/endpoints/organization_spans_aggregation.py

```diff
@@ -0,0 +1,230 @@
+                    spans_dict["coalesced_group"] = (
+                        spans_dict["group_raw"]
+                        if spans_dict["group"] == NULL_GROUP
+                        else spans_dict["group"]
+                    )
+                    span_tree[span_id] = spans_dict
+                    span_tree[span_id]["children"] = []
+
+            for span in span_tree.values():
+                parent_id = span["parent_span_id"]
+                try:
+                    parent_span = span_tree[parent_id]
+                    children = parent_span["children"]
+                    children.append(span)
+                except KeyError:
+                    pass
+
+            try:
+                root_span = span_tree[root_span_id]
+            except KeyError:
+                continue
+
+            self.fingerprint_nodes(root_span, root_span["start_ms"])
+
+        return Response(data=self.aggregated_tree)
+
+    def fingerprint_nodes(
+        self,
+        span_tree,
+        root_start_timestamp,
+        root_prefix=None,
+        parent_node_fingerprint=None,
+        nth_span=0,
+    ):
+        """
+        Build a fingerprint using current span group and span groups of all the spans in the path before it.
+        Example 1:
+            A
+            |--B
+            |--C
+               |--D
+
+            In this example, given A, B, C, D are span groups, the fingerprint of span D would be
+            the md5 hash of the value A-C-D.
+
+        Example 2:
+            A
+            |--B
+            |--C
+            |  |--D
+            |
+            |--C
+               |--E
+
+            In this example, given A, B, C, D are span groups, the fingerprint of span D would be
+            the md5 hash of the value A-C-D and for span E would be md5 hash of the value A-C1-E.
+        """
+        coalesced_group = span_tree["coalesced_group"]
+        description = span_tree["description"]
+        start_ms = span_tree["start_ms"]
+        if root_prefix is None:
+            prefix = coalesced_group
+        else:
+            if nth_span == 0:
+                prefix = f"{root_prefix}-{coalesced_group}"
+            else:
+                prefix = f"{root_prefix}-{coalesced_group}{nth_span}"
+        node_fingerprint = hashlib.md5(prefix.encode()).hexdigest()[:16]
+
+        try:
+            node = self.aggregated_tree[node_fingerprint]
+            count = node["count()"]
+            node["avg(exclusive_time)"] = incremental_average(
+                node["avg(exclusive_time)"], count, span_tree["exclusive_time"]
+            )
+            node["avg(duration)"] = incremental_average(
+                node["avg(duration)"], count, span_tree["duration"]
+            )
+            node["avg(offset)"] = incremental_average(
+                node["avg(offset)"],
+                count,
+                start_ms - root_start_timestamp,
+            )
+            node["count()"] += 1
+        except KeyError:
```

Comment:
Related to comment above, could we make this an explicit check?

Reply:
https://github.com/getsentry/sentry/pull/56291/commits/1fbbfe05516aba2dd890cb846b44a7af04a3de77#diff-1df1994c8fddccf95629e8b491f1ff17c43534b3634dd396256fc448124d7cddR177